applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -894,29 +895,19 @@ void BaseSolidElement::CalculateOnIntegrationPoints(
                 // Compute material reponse
                 CalculateConstitutiveVariables(this_kinematic_variables, this_constitutive_variables, Values, point_number, integration_points, GetStressMeasure());
 
-                const Matrix stress_tensor = MathUtils<double>::StressVectorToTensor( this_constitutive_variables.StressVector );
-
-                double sigma_equivalent = 0.0;
-
                 if (dimension == 2) {
-
-                    sigma_equivalent = std::pow(stress_tensor(0,0),2) 
-                                     + std::pow(stress_tensor(1,1),2) 
-                                     - stress_tensor(0,0)*stress_tensor(1,1) 
-                                     + 3*(stress_tensor(0,1) * stress_tensor(1,0));
+                    array_1d<double, 3> stress_vector;
```

Comment:
@loumalouomega why don't you template the function in the vector type? this way you can avoid this copy.

Reply:
Is for the original CL which uses array